shell/platform/darwin/macos/framework/Source/FlutterSurface.mm

```diff
@@ -0,0 +1,94 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterSurface_Internal.h"
+
+#import <Metal/Metal.h>
+
+@interface FlutterSurface () {
+  CGSize _size;
+  IOSurfaceRef _ioSurface;
+  id<MTLTexture> _texture;
+}
+@end
+
+@implementation FlutterSurface
+
+- (IOSurfaceRef)ioSurface {
+  return _ioSurface;
+}
+
+- (CGSize)size {
+  return _size;
+}
+
+- (int64_t)textureId {
+  return reinterpret_cast<int64_t>(_texture);
+}
+
+- (instancetype)initWithSize:(CGSize)size device:(id<MTLDevice>)device {
```

Comment:
Is it the Objective-C style to hide these methods in the `.mm` although they're used by other classes? I don't think it's necessary to make `FlutterSurface` so opaque since it's only used in the embedding anyway. Rather, it's much clearer to show which methods are intended to be used by other internal classes.